approval_polls/views.py

```diff
@@ -36,6 +37,12 @@ class DetailView(generic.DetailView):
   def get_queryset(self):
       return Poll.objects.filter(pub_date__lte=timezone.now())
 
+@login_required
+def delete_poll(request):
```

Comment:
This should not have a new route name like 'delete_poll' but should be part of the detail view. It will be similar to the [CreateView class](https://github.com/electology/approval_frame/blob/master/approval_polls/views.py#L63) but instead of get and post it will have the methods get and delete.